src/common/utils.ts

```diff
@@ -102,42 +103,100 @@ const transformTokenRequestSnakeCase = (
   } as Token | CreateToken | UpdateToken;
 };
 
-const transformTokenResponseCamelCase = (token: Token): Token | undefined => {
-  if (typeof token === 'undefined') {
+// any as we dont know if its a list or not
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isList = <T>(arg: any): arg is PaginatedList<T> =>
+  arg && arg?.pagination !== undefined && arg?.data !== undefined;
+
+const transformTokenResponseCamelCase = (
+  tokenResponse: Token | PaginatedList<Token>
+): Token | PaginatedList<Token> | undefined => {
+  if (typeof tokenResponse === 'undefined') {
     return undefined;
   }
 
+  if (isList<Token>(tokenResponse)) {
+    const transformedData = tokenResponse.data.map((t: Token) => ({
+      ...camelcaseKeys(t, { deep: true }),
+      ...(t.data !== undefined ? { data: t.data } : {}),
+      ...(t.metadata !== undefined ? { metadata: t.metadata } : {}),
+    }));
+
+    const transformedToken = {
+      data: transformedData,
+      pagination: camelcaseKeys(tokenResponse.pagination, { deep: true }),
+    };
+
+    return transformedToken;
+  }
+
   return {
-    ...camelcaseKeys(token, { deep: true }),
-    ...(token.data !== undefined ? { data: token.data } : {}),
-    ...(token.metadata !== undefined ? { metadata: token.metadata } : {}),
+    ...camelcaseKeys(tokenResponse, { deep: true }),
+    ...(tokenResponse.data !== undefined ? { data: tokenResponse.data } : {}),
+    ...(tokenResponse.metadata !== undefined
+      ? { metadata: tokenResponse.metadata }
+      : {}),
   } as Token;
 };
 
 const transformReactorResponseCamelCase = (
-  reactor: Reactor
-): Reactor | undefined => {
-  if (typeof reactor === 'undefined') {
+  reactorResponse: Reactor | PaginatedList<Reactor>
+): Reactor | PaginatedList<Reactor> | undefined => {
+  if (typeof reactorResponse === 'undefined') {
```

Comment:
Same here, should we update the functions type signature to reflect that `reactorResponse` can also be `undefined`?